app/models/variant.rb

```diff
@@ -94,8 +94,15 @@ def active?
   end
 
   # This is a form helper to inactivate a variant
+  #
+  # if :inactivate checkbox is checked (== '1') for the first time (!deleted_at),
+  # record :deleted_at time. if it isn't checked or was unchecked, make it .active? again.
   def inactivate=(val)
-    self.deleted_at = Time.zone.now if !deleted_at && (val && (val == '1' || val.to_s == 'true'))
+    if val == '1' 
+      self.deleted_at = Time.zone.now if !deleted_at
+    else
+      self.deleted_at = nil
```

Comment:
I don't think the `inactivate=(val)` should be setting this to active.  I need to test the UI but the method's name either needs to change or the UI might need changing.